### PR TITLE
LPS-177939 Include FF on the back story and save the user agent when active

### PR DESCRIPTION
**Pull request by AliciaGarciaGarcia**
This is a subsequent PR with changes on the backend to give the user the ability to save the user agent if FF is active

FF : LPS-175850

- LPS-177939 upgrade the old configurations to be for all the users
- LPS-177939 only apply the new logic if the FF is active
- LPS-177939 on the case the FF is active, the pattern string should include the user agent
- LPS-177939 update test depending the FF
- LPS-177939 modify how we update the configuration of the patterns
- LPS-177939 baseline
- LPS-177939 if the FF is enabled obtain the user agent from the page
- LPS-177939 put the user agent on the element

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
ci:stop

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
ci:stop

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
this PR is working strange T_T

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Comment by adolfopa** [comment]
ci:reevaluate:1407401_1059

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
ci:reevaluate:1407401_1085

File: modules/apps/redirect/redirect-service/src/main/java/com/liferay/redirect/internal/util/PatternUtil.java [hunk 1]
```diff
@@ -29,19 +31,35 @@ public class PatternUtil {
 	public static List<RedirectPatternEntry> parse(String[] patternStrings) {
 		List<RedirectPatternEntry> redirectPatternEntries = new ArrayList<>();
 
+		int minLength = 2;
+
+		if (FeatureFlagManagerUtil.isEnabled("LPS-175850")) {
+			minLength = 3;
+		}
+
```
Comment (adolfopa) [inline modules/apps/redirect/redirect-service/src/main/java/com/liferay/redirect/internal/util/PatternUtil.java:39]: Can you call this "limit"? Calling it "min" implies that there's a "max".

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: modules/apps/redirect/redirect-service/src/test/java/com/liferay/redirect/internal/provider/RedirectProviderImplTest.java [hunk 2]
```diff
@@ -49,6 +53,7 @@ public class RedirectProviderImplTest {
 
 	@Before
 	public void setUp() {
+		MockitoAnnotations.initMocks(this);
```
Comment (adolfopa) [inline modules/apps/redirect/redirect-service/src/test/java/com/liferay/redirect/internal/provider/RedirectProviderImplTest.java:56]: Shouldn't there be a newline after this?

File: modules/apps/redirect/redirect-service/src/test/java/com/liferay/redirect/internal/provider/RedirectProviderImplTest.java [hunk 3]
```diff
@@ -250,6 +269,11 @@ public void testSimplePatternDoesntMatchUserAgentHuman() {
 
 	@Test
 	public void testSimplePatternMatchesNoUserAgent() {
+		Mockito.when(
+			_props.get("feature.flag.LPS-175850")
+		).thenReturn(
+			"true"
+		);
```
Comment (adolfopa) [inline modules/apps/redirect/redirect-service/src/test/java/com/liferay/redirect/internal/provider/RedirectProviderImplTest.java:276]: Newline after the mock expectations.

File: modules/apps/redirect/redirect-service/src/test/java/com/liferay/redirect/internal/provider/RedirectProviderImplTest.java [hunk 4]
```diff
@@ -265,6 +289,11 @@ public void testSimplePatternMatchesNoUserAgent() {
 
 	@Test
 	public void testSimplePatternMatchesNoUserAgentOnRedirect() {
+		Mockito.when(
+			_props.get("feature.flag.LPS-175850")
+		).thenReturn(
+			"true"
+		);
```
Comment (adolfopa) [inline modules/apps/redirect/redirect-service/src/test/java/com/liferay/redirect/internal/provider/RedirectProviderImplTest.java:296]: Newline.

File: modules/apps/redirect/redirect-service/src/test/java/com/liferay/redirect/internal/provider/RedirectProviderImplTest.java [hunk 5]
```diff
@@ -319,6 +400,11 @@ public void testSimplePatternSingleMatch() {
 
 	@Test
 	public void testSimplePatternSingleMatchesUserAgentAll() {
+		Mockito.when(
+			_props.get("feature.flag.LPS-175850")
+		).thenReturn(
+			"true"
+		);
```
Comment (adolfopa) [inline modules/apps/redirect/redirect-service/src/test/java/com/liferay/redirect/internal/provider/RedirectProviderImplTest.java:407]: Newline.

File: modules/apps/redirect/redirect-service/src/test/java/com/liferay/redirect/internal/provider/RedirectProviderImplTest.java [hunk 6]
```diff
@@ -335,6 +421,11 @@ public void testSimplePatternSingleMatchesUserAgentAll() {
 
 	@Test
 	public void testSimplePatternSingleMatchesUserAgentHuman() {
+		Mockito.when(
+			_props.get("feature.flag.LPS-175850")
+		).thenReturn(
+			"true"
+		);
```
Comment (adolfopa) [inline modules/apps/redirect/redirect-service/src/test/java/com/liferay/redirect/internal/provider/RedirectProviderImplTest.java:428]: Newline.

File: modules/apps/redirect/redirect-service/src/test/java/com/liferay/redirect/internal/util/PatternUtilTest.java [hunk 7]
```diff
@@ -74,13 +94,27 @@ public void testCaretSlashPattern() {
...
 		Assert.assertTrue(
 			ListUtil.isEmpty(PatternUtil.parse(new String[] {"xyz "})));
 		Assert.assertTrue(
 			ListUtil.isEmpty(PatternUtil.parse(new String[] {"xyz"})));
+
+		Mockito.when(
+			_props.get("feature.flag.LPS-175850")
+		).thenReturn(
+			"true"
+		);
```
Comment (adolfopa) [inline modules/apps/redirect/redirect-service/src/test/java/com/liferay/redirect/internal/util/PatternUtilTest.java:109]: Newline.

**Review by adolfopa — CHANGES_REQUESTED** [review]

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Review by adolfopa — APPROVED** [review]

**Comment by adolfopa** [comment]
@AliciaGarciaGarcia Sorry, rebase error. But as tests passed and the pr is approved, you may try to forward it directly once fixed. No need to wait again for all tests to pass.

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:forward

**Comment by AliciaGarciaGarcia** [comment]
ci:forward

**Comment by AliciaGarciaGarcia** [comment]
ci:forward

**Comment by AliciaGarciaGarcia** [comment]
ci:forward

**Pull request closed without merging**